package/Dockerfile

```diff
@@ -1,38 +1,77 @@
 
-ARG kubectl_version=1.28.0
-ARG ARCH
+ARG TARGETARCH
 
 # Install system packages using builder image that has zypper 
 COPY --from=micro / /chroot/
 
-# Install kubectl into micro
-RUN curl -Lo /chroot/usr/local/bin/kubectl "https://dl.k8s.io/release/v${kubectl_version}/bin/linux/${ARCH}/kubectl" && chmod +x /chroot/usr/local/bin/kubectl
+# OS binaries to run kube-bench audit commands.
+# Before removing dependencies here, ensure they are not needed by cis-operator,
+# security-scan or any of the existing kube-bench tests.
+RUN zypper --installroot /chroot -n --gpg-auto-import-keys in --no-recommends findutils tar jq gawk diffutils procps systemd gzip curl && \
+    zypper --installroot /chroot clean -a && \
+    rm -rf /chroot/var/cache/zypp/* /chroot/var/log/zypp/*
+
+
+# Stage Sonobuoy into builder
+ENV SONOBUOY_SUM="SONOBUOY_SUM_${TARGETARCH}"
+RUN curl --output /tmp/sonobuoy.tar.gz -sLf "https://github.com/vmware-tanzu/sonobuoy/releases/download/v${SONOBUOY_VERSION}/sonobuoy_${SONOBUOY_VERSION}_linux_${TARGETARCH}.tar.gz" && \
+    echo "${!SONOBUOY_SUM}  /tmp/sonobuoy.tar.gz" | sha256sum -c - && \
+    tar -xvzf /tmp/sonobuoy.tar.gz -C /chroot/usr/bin sonobuoy
+
+# Stage kubectl into builder
+ADD --chown=root:root --chmod=0755 \
+    "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl" \
+    /chroot/usr/local/bin/kubectl
 
-## Install Sonobuoy into micro
-RUN curl -sLf "https://github.com/vmware-tanzu/sonobuoy/releases/download/v${sonobuoy_version}/sonobuoy_${sonobuoy_version}_linux_${ARCH}.tar.gz" | tar -xvzf - -C /chroot/usr/bin sonobuoy
+ENV KUBECTL_SUM="KUBECTL_SUM_${TARGETARCH}"
+RUN echo "${!KUBECTL_SUM}  /chroot/usr/local/bin/kubectl" | sha256sum -c -
 
-## Install kube-bench into micro
-RUN curl -sLf "https://github.com/aquasecurity/kube-bench/releases/download/v${kube_bench_version}/kube-bench_${kube_bench_version}_linux_${ARCH}.tar.gz" | tar -xvzf - -C /chroot/usr/bin
+# Stage kube-bench into builder
+ENV KUBE_BENCH_SUM="KUBE_BENCH_SUM_${TARGETARCH}"
+RUN curl --output /tmp/kubebench.tar.gz -sLf "https://github.com/aquasecurity/kube-bench/releases/download/v${KUBE_BENCH_VERSION}/kube-bench_${KUBE_BENCH_VERSION}_linux_${TARGETARCH}.tar.gz" && \
+    echo "${!KUBE_BENCH_SUM}  /tmp/kubebench.tar.gz" | sha256sum -c - && \
+    tar -xvzf /tmp/kubebench.tar.gz -C /chroot/usr/bin
 
-## Copy the files within /cfg straight from the immutable GitHub source to /etc/kube-bench/cfg/ into micro
+# Copy the files within /cfg straight from the immutable GitHub source to /etc/kube-bench/cfg/ into micro
 RUN mkdir -p /chroot/etc/kube-bench/ && \
-    curl -sLf "https://github.com/aquasecurity/kube-bench/archive/refs/tags/v${kube_bench_version}.tar.gz" | \
-    tar xvz -C /chroot/etc/kube-bench/ --strip-components=1 "kube-bench-${kube_bench_version}/cfg"
+    curl -sLf "https://github.com/aquasecurity/kube-bench/archive/refs/tags/v${KUBE_BENCH_VERSION}.tar.gz" | \
```

Comment:
Source code is immutable in GH releases, so unless the tag/release has been deleted and recreated, this should not change.

We should handle this separately - on a follow-up PR. An idea would be to download and sync upstream cfgs to our configs, so that we can go through the same validation processes we undergo for our own.

Reply:
cc @andypitcher wdyt?

Reply:
This is a good idea, this could be the beginning of automating the upgrade of our CIS profiles ? As soon as a new version of upstream CFGs is downloaded, new PRs are automatically created to make rke/k3s/rke2 (hardened/permissive) profiles. That might require `renovate`, maybe?

Reply:
We probably could do this independently of renovate. Maybe a new Make target that does that op, and a GHA workflow that creates a PR when it identifies changes on the worktree.